src/as3hx/Writer.hx

```diff
@@ -2565,10 +2565,17 @@ class Writer
                     var rebuiltExpr = EField(EIdent("Std"), "string");
                     rebuiltCall = ECall(rebuiltExpr, [e]);
                 }
+                else if (f == "concat" && params.empty()) {
+                    var type = getExprType(e);
+                    if (type != null && type.indexOf("Array") != -1) {
+                        var rebuildExpr = EField(e, "copy");
+                        rebuiltCall = ECall(rebuildExpr, params);
+                    }
```

Comment:
I've done similar to the indexOf. If in the future there will be problems, then I will fix them